doc_src/function.txt

```diff
@@ -29,7 +29,7 @@ The following options are available:
 
 - `-s` or `--on-signal SIGSPEC` tells fish to run this function when the signal SIGSPEC is delivered. SIGSPEC can be a signal number, or the signal name, such as SIGHUP (or just HUP).
 
-- `-B` or `--shadow-builtin` must be specified if the function name is the same as a builtin. Specifying this flag indicates your acknowledgement that you are wrapping or replacing the builtin command. This is a safety feature to make it harder for people to inadvertently break the shell by doing things like `function test; return 0; end`. If the function name is not currently a builtin using this flag will produce an error. If you want to write a function that provides a builtin to an older version of fish you need to add something like `builtin --names | grep -q '^cmd$'; and return` to the top of the function script (where `cmd` is the name of the builtin/function). That will keep your script from replacing the builtin with your function on the newer fish version while allowing your function to provide similar functionality on older versions of fish.
+- `-B` or `--shadow-builtin` should be specified if the function name is the same as a builtin. Specifying this flag indicates your acknowledgement that you are wrapping or replacing the builtin command. This is a safety feature to make it harder for people to inadvertently break the shell by doing things like `function test; return 0; end`. If you do that you'll get a warning message written to stderr but your function will still be created. If the function name is not currently a builtin using this flag will produce an error and the function will not be created. If you want to write a function that provides a builtin to an older version of fish you need to add something like `builtin --names | grep -q '^cmd$'; and return` to the top of the function script (where `cmd` is the name of the builtin/function). That will keep your script from replacing the builtin with your function on the newer fish version while allowing your function to provide similar functionality on older versions of fish.
```

Comment:
I would reword this text as follows:

`-B` or `--shadow-builtin` silences the warning that is produced when the function name is the same as a builtin. It should be specified when replacing or wrapping a builtin command. This is a safety feature to maintain the integrity of the environment; a common mistake is to create a function called `test`, when much of the shell internals rely on the behaviour of the existing test builtin. Without this flag, a warning message will be printed, although the function will still be created. Note that, if the function name is not builtin, this flag will produce an error and the function will not be created. To create fallback functions for old versions of fish, consider using a test like `contains function_name (builtin --names)` before defining the function.